packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -171,7 +171,28 @@ class RefreshIndicator extends StatefulWidget {
     this.semanticsValue,
     this.strokeWidth = RefreshProgressIndicator.defaultStrokeWidth,
     this.triggerMode = RefreshIndicatorTriggerMode.onEdge,
-  }) : _indicatorType = _IndicatorType.adaptive;
+  }) : _indicatorType = _IndicatorType.adaptive, onModeChange = null;
+
+  /// Creates [RefreshIndicator] with no spinner and calls `onRefresh` when
+  /// successfully armed by drag event.
+  const RefreshIndicator.noSpinner({
+    super.key,
+    required this.child,
+    required this.onRefresh,
+    this.onModeChange,
```

Comment:
I suppose `onModeChange` could equally well be an ordinary RefreshIndicator parameter. On the other hand, that's something we can easily add in the future and I appreciate the narrow scope of this PR.

Reply:
Lets address this in a separate PR 💪